packages/dappmanager/src/ethForward/index.ts

```diff
@@ -1,137 +1,79 @@
- * - decentral.eth => true
- * - my.dappmanager.dnp.dappnode.eth => false
- * - my.dappnode => false
- */
-function isDwebRequest(req: http.IncomingMessage): boolean {
-  const domain = req.headers.host;
-  return (
-    typeof domain === "string" &&
-    domain.endsWith(".eth") &&
-    !domain.endsWith("dnp.dappnode.eth")
-  );
-}
-
-/**
- * Start eth forward http proxy
- */
-function getEthForwardHandler(): (
-  req: http.IncomingMessage,
-  res: http.ServerResponse
-) => Promise<void> {
-  logs.info(`IPFS redirect set to: ${params.ETHFORWARD_IPFS_REDIRECT}`);
-  logs.info(`SWARM redirect set to: ${params.ETHFORWARD_SWARM_REDIRECT}`);
-
-  const proxy = httpProxy.createProxyServer({});
-  proxy.on("error", e => {
-    logs.error(`ETHFORWARD proxy error`, e);
-  });
-
   // Create a domain resolver with cache
   const resolveDomain = ResolveDomainWithCache();
+  // Start ethforward http proxy: Resolves .eth domains
+  const ethForwardHandler = getIpfsProxyHandler<string>(
+    "ETHFORWARD",
+    async (_req, domain) => await resolveDomain(domain)
+  );
 
-  return async function ethForwardProxyHttpHandler(
-    req: http.IncomingMessage,
-    res: http.ServerResponse
-  ): Promise<void> {
-    const domain = req.headers.host;
-    try {
-      if (!domain) throw new TypeError(`req host is not defined`);
-
-      const content = await resolveDomain(domain);
-      const target = getTargetUrl(content);
-      logs.debug(`Proxying ${domain} to ${target}`, content);
+  // Start IPFS gateway proxy: Serve IPFS content from the DAPPMANAGER
+  const ipfsGatewayProxydHandler = getIpfsProxyHandler<string>(
+    "IPFS gateway",
+    async (_req, hash) => ({ location: "ipfs", hash })
+  );
 
-      // Must change the host from something not *.eth, to prevent the IPFS node
-      // from trying to resolve the domain with DNSLink and causing an error
-      req.headers.host = "dappmanager.dappnode";
+  return (req, res, next): void => {
+    try {
+      const domain = parseEthDomainHost(req);
+      if (domain !== null) {
+        ethForwardHandler(req, res, domain);
+        return;
+      }
 
-      // Note: Must use the promise constructor otherwise proxy.web breaks
-      // due to no proper binding of its underlying 'this'
-      //   Cannot read property 'length' of undefined
-      //   src/node_modules/http-proxy/lib/http-proxy/index.js:72:31
-      await new Promise<http.IncomingMessage>((resolve, reject) => {
-        proxy.web(req, res, { target }, (e: Error | undefined, proxyRes) => {
-          if (!e) return resolve(proxyRes);
-          // Indicates that the host is unreachable.
-          // Usually happens when the node is not running
-          if (e.message.includes("EHOSTUNREACH"))
-            reject(new NodeNotAvailable(e.message, content.location));
-          else reject(new ProxyError(e.message, target));
-        });
-      });
+      const hash = parseIpfsGatewayProxyReqHash(req.url);
```

Comment:
boolean function to check if is hash or not ?

Reply:
Same reasoning as with the domain. The goal is to not have to repeat work and only parse the hash once.